tests/flytekit/unit/core/test_type_hints.py

```diff
@@ -1338,3 +1340,16 @@ def t2() -> dict:
     expected_struct = Struct()
     expected_struct.update({"k1": "v1", "k2": 3, "4": {"one": [1, "two", [3]]}})
     assert output_lm.literals["o0"].scalar.generic == expected_struct
+
+
+def test_error_messages():
+    @task
+    def foo(a: int, b: str) -> typing.Tuple[int, str]:
+        return 10, "hello"
+
+    @task
+    def foo2(a: int, b: str) -> typing.Tuple[int, str]:
+        return "hello", 10
+
+    with pytest.raises(TypeError):
```

Comment:
can you also use the `match` parameter for this test? Just to confirm that the message is what we expect.